### DISCUSSION TITLE
Changing the license of the code examples in the docs to public domain "CC0 1.0 universal"

### DISCUSSION THREAD

**Discussion by Isam-Lakehal** [nodejs/help#4670] [Polls]
Do you think that the code examples in the docs should be licensed under the public domain, thus anyone can learn from them without worrying that he might use the same code in his projects and violate the copyright?

**Comment by avivkeller** [comment]
@nodejs/tsc - The TSC handles copyrights and licenses, correct?

**Comment by aduh95** [comment]
Can you clarify what code are you talking about and what limitation of the current license bothers you?

**Reply by tniessen** [reply]
Not a lawyer obviously, but I think individual examples in our docs probably don't meet the threshold of originality in most jurisdictions.

**Reply by Isam-Lakehal** [reply]
Mozilla released a lot of HTML CSS and JS code examples to the public domain, and I think this eliminates the headache of attribution and including the license whenever using that code in any project, plus as a beginner it's hard to avoid using almost the same code whenever you approach the same problem, once you see the code it sticks to your mind. in general I think this should be the norm for all code examples in any documentation.

**Reply by avivkeller** [reply]
Doesn't the documentation fall under the project's main license?

If so, isn't using examples from the documentation (license-wise) the same as using Node.js to run a project? Why would the examples require a special exception, when you'd use the licensed `node` source to run them?

> (I'm also not a lawyer so I might be misunderstanding something)

**Reply by benjamingr** [reply]
@RedYetiDev because you're using the Node.js binary as a binary but using the actual code from the documentation

**Comment by benjamingr** [comment]
I think OP is right here in terms of legalese, I agree with @tniessen that examples likely don't meet the threshold of originality and mostly - I don't think there is a way for us to release the MIT code as anything else since we don't own it (we'd have to go to every contributor of the code examples and ask for permission)

**Reply by Isam-Lakehal** [reply]
I agree that asking every contributor would be a tedious task  but it's doable if there is a will for that?
I'd volonteer for this task with whomever wants to do the same.

**Reply by benjamingr** [reply]
Thousands of people have contributed to Node.js over the years. If the docs have 1/10th of that (assuming the code examples aren't copied from tests authored by other people for example) that would still be contacting and getting legal permission to relicense from over 300 people. Many of those people are inactive in the project or more generally on the internet - some may be dead and you'd need permission from their estates etc.

I'm happy to license any of _my_ docs contributions under public domain (like [I do](https://stackoverflow.com/users/1348195/benjamin-gruenbaum?tab=profile) with my SO answers) but without every contributor involved I doubt that's a realistic undertaking.

If you feel strongly about this, we can ask for guidance from our lawyers at the OpenJS foundation, but I doubt they'll somehow let us relicense those examples and if you don't believe those examples don't meet the threshold of originality I doubt OpenJS lawyers saying that that's their opinion would help much. I happily use them in my code but I'm not a lawyer and your concerns are definitely legitimate.

Another option is to include attribution if you're worried. MIT is a very permissive open source license that lets you use whatever code you want and modify it how you want as long as you include the license.

**Reply by Isam-Lakehal** [reply]
You have a point there, probably getting all code examples under a public domain license would not be feasible, but getting the majority of them might be, I suggest doing as I think mozilla did, by licensing new contributions under CC0 and creating new repositories where people can contribute their already published code examples or contribute new examples and licenesing them under CC0, team effort is the key here.

**Reply by mcollina** [reply]
I'm not opposed if:

1. there are (new) volunteers to handle it on our side
2. there is a donor that covers the lawyer's costs (this likely measures in 10s of thousands)

A more lightweight option could be to do it for the Learn section of the website: https://nodejs.org/learn/getting-started/introduction-to-nodejs.

Moreover I think this would be a distraction for the project, as we have more pressing priorities.

**Reply by Isam-Lakehal** [reply]
@mcollina I didn't know that it would cost this much and also involves lawyers, what I had in mind is that if someone contributed an example, we just need to ask him if he wants to release it to the public domain and if he refuses maybe there will be someone willing to write an example for that particular subject and contribute it. So since this is the case it would make it even harder.

Doing it for the learning section would be a great start, what I also suggest if it's feasible is creating a repo with the same structure as the docs, and inviting new collaborators and letting each one contribute with whatever he thinks he can give a good example about.

Count me as a volunteer for this project.
